lib/fetch/body.js

```diff
@@ -443,14 +445,13 @@ function bodyMixinMethods (instance) {
           let text = ''
           // application/x-www-form-urlencoded parser will keep the BOM.
           // https://url.spec.whatwg.org/#concept-urlencoded-parser
-          const textDecoder = new TextDecoder('utf-8', { ignoreBOM: true })
           for await (const chunk of consumeBody(this[kState].body)) {
             if (!isUint8Array(chunk)) {
               throw new TypeError('Expected Uint8Array chunk')
             }
-            text += textDecoder.decode(chunk, { stream: true })
+            text += textDecoderNoBom.decode(chunk, { stream: true })
```

Comment:
Is it safe to reuse the decoder in stream mode for multiple requests?

Reply:
Good question. Documentation says this:

`Decodes the input and returns a string. If options.stream is true, any incomplete byte sequences occurring at the end of the input are buffered internally and emitted after the next call to textDecoder.decode().`

That does make it sound like streaming decoding is, indeed, stateful.

@mcollina any comments on this?

Reply:
Yea, don't re-use in the streaming case.

Reply:
gotcha, will adjust the streaming case shortly

Reply:
fixed